### PR TITLE
Insert arguments into types SDB on function analysis ##anal

### PR DESCRIPTION
**Pull request by oxcabe**
This is helpful for generating and matching types when using zignatures.

### CODE REVIEW

**Comment by oxcabe** [comment]
Don't merge yet anyway, I'm working to fix something regarding this PR atm

**Comment by radare** [comment]
use only one ##

**Comment by oxcabe** [comment]
Can someone give opinions on this so far? 

Tests are also fixed now:
https://github.com/radareorg/radare2-regressions/pull/2007

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -815,6 +815,29 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+	char *query = NULL;
+	int arg_count = 0;
+
+	RList *all_vars = cache.bvars;
+	r_list_join (all_vars, cache.svars);
+	r_list_join (all_vars, cache.rvars);
+
+	r_list_foreach (all_vars, iter, var) {
+		if (var->isarg) {
+			query = r_str_newf ("anal/types/func.%s.arg.%d=%s,%s", fcn->name, arg_count, var->type, var->name);
```
Comment (radare) [inline libr/core/cmd_anal.c]: its leaking, use sdb_fmt or free that query after using it

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -815,6 +815,29 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+			sdb_querys (core->sdb, NULL, 0, query);
+			++arg_count;
```
Comment (radare) [inline libr/core/cmd_anal.c]: never use preincrements, they are totallyh unnecessary the 99% of the time

Comment (oxcabe) [reply libr/core/cmd_anal.c]: TIL :smiley:

**Pull request merged**